pkg/alertmanager/operator.go

```diff
@@ -920,6 +920,23 @@ receivers:
 	}
 
 	generator := newConfigGenerator(namespacedLogger, version, store)
+	// If defined AlertmanagerConfiguration, find the global AlertmanagerConfig and generate.
+	// Otherwise work with the raw configuration secret.
+	if am.Spec.AlertmanagerConfiguration != nil {
+		globalAmConfig, err := c.mclient.MonitoringV1alpha1().AlertmanagerConfigs(am.Namespace).
+			Get(ctx, am.Spec.AlertmanagerConfiguration.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrap(err, "get AlertmanagerConfig failed")
+		}
+		baseConfig, err = generator.generateGlobalConfig(ctx, globalAmConfig)
+		if err != nil {
+			return errors.Wrap(err, "failed to generate global AlertmangerConfig")
+		}
+		if err := checkAlertmanagerConfigRootRoute(baseConfig.Route); err != nil {
+			return errors.Wrap(err, "check AlertmanagerConfig root route failed")
+		}
```

Comment:
This should be moved to `generator.generateGlobalConfig()` IMHO.

Reply:
Yes, fixed at [292dd07](https://github.com/prometheus-operator/prometheus-operator/pull/4220/commits/292dd07617f2e310908af5aa0cf0fa4edc950c06)